apps/user_ldap/lib/Jobs/UpdateGroups.php

```diff
@@ -79,28 +114,35 @@ public function updateGroups() {
 	}
 
 	/**
-	 * @return int
+	 * @return array
 	 */
-	private function getRefreshInterval() {
-		//defaults to every hour
-		return \OC::$server->getConfig()->getAppValue('user_ldap', 'bgjRefreshInterval', 3600);
+	private function getKnownGroups() {
+		if (is_array($this->groupsFromDB)) {
+			$this->groupsFromDB;
+		}
+		$qb = $this->dbc->getQueryBuilder();
+		$qb->select(['owncloudname', 'owncloudusers'])
+			->from('ldap_group_members');
+		$result = $qb->execute()->fetchAll();
+
+		$this->groupsFromDB = [];
+		foreach ($result as $dataset) {
+			$this->groupsFromDB[$dataset['owncloudname']] = $dataset;
```

Comment:
```suggestion
			$this->groupsFromDB[$dataset['owncloudname']][]= $dataset;
```
or some other way of merging results? 
Or do you really only care about the last owncloudname user being added? In that case add a group by?

Reply:
Or is owncloudname a unique key?

Reply:
> Or is owncloudname a unique key?

yes, maps to the groupId